content/markdown/02-working-with-data/02-05-new-columns.rmd

```diff
@@ -0,0 +1,151 @@
+df = lterdatasampler::hbr_maples
+head(df)
+```
+For this example, we only will work with some of the variables, so let's first select the columns we want (LINK TO SECTION ON SELECTION). 
+
+```{r}
+df = df %>% 
+  dplyr::select(year, watershed, elevation, leaf1area, 
+                leaf2area, corrected_leaf_area)
+df
+```
+
+
+We could for example, want a column in this dataset that tells us what forest these data are from. That would be an easy use of the `mutate()` function as there's only one value. We could do this like so:
+
+```{r}
+df = df %>% 
+  dplyr::mutate(forest = "HBEF")
+df
+```
+
+And we see that our new column has been added onto the far right of the dataframe. 
+
+## Mutating Columns as a Function of Existing Columns
+
+A more common use case is if we want to make a new column from existing columns. Two typical uses here would be:
+
+1. Making a new column that is made via a function operating on another column(s)
+
+2. Making a factor column that binds together values from other columns 
+
+### 1 - New Column Via Numeric Function or Logical Test
+
+#### Rowwise Function
+
+In these data, there are two measures of leaf area: `leaf1area` and `leaf2area`. While they're both very similar, it might still be useful to have a mean estimate for each observation in the data between the two leaf area measurements. 
+
+Say we wanted to calculate for row, the mean value of the two measures of leaf area. **To make this operation on a row-wise basis, we can use `dplyr::rowwise()`** which will ensure each calculation is done on a single row. Combining this with `mutate()` will produce our desired result. 
+
+```{r}
+df = df %>% 
+  dplyr::rowwise() %>%  # ensures calculation done for each row
+  dplyr::mutate(
+    mean_area = mean(c(leaf1area, leaf2area), na.rm = TRUE)
+  )
+df
+```
+
+Now, again, we have our new column tacked on at the end of our dataframe just as we wanted. 
+
+Since we have now a mean estimate of leaf area, we might want to compare that to our measure of `corrected_leaf_area`. We might want, for instance, to see if the `corrected_leaf_area` is a consistent proportion larger than the mean of the two leaf area measurements.
+
+To go about this, we might want to get a ratio of the `mean_area` and the `corected_leaf_area` for each row. To do so, we can create an operation very similarly to above: 
+
+```{r}
+df = df %>% 
+  dplyr::rowwise() %>%  # we still need this to get a value for each row
+  dplyr::mutate(
+    corrected_mean_ratio = mean_area / corrected_leaf_area
+  )
+df
+```
+
+#### ifelse()
+
+A handy component of `mutate()` is that we can create a new column based on a conditional statement like an `ifelse()` statement (LINK TO PROGRAMMING CONCEPT OF IFELSE). In the `corrected_leaf_area` column, we may want to know if a particular value falls above or below the mean value for all of our observations. We could create a simple categorical variable based on this for each row with an `ifelse()` statement. 
```

Comment:
It's a good idea to link to the concept of ifelse here. It might also be worth mentioning briefly a bit more about this concept here with the link. Even a sentence might be helpful.